functions/billing/main.py

```diff
@@ -42,26 +44,36 @@
 # [END functions_billing_limit]
 
 # [START functions_billing_slack]
-
 # See https://api.slack.com/docs/token-types#bot for more info
 BOT_ACCESS_TOKEN = 'xxxx-111111111111-abcdefghidklmnopq'
-
-CHANNEL_ID = 'C0XXXXXX'
+CHANNEL = 'C0XXXXXX'
 
 slack_client = slack.WebClient(token=BOT_ACCESS_TOKEN)
 
 
 def notify_slack(data, context):
     pubsub_message = data
 
-    notification_attrs = json.dumps(pubsub_message['attributes'])
-    notification_data = base64.b64decode(data['data']).decode('utf-8')
-    budget_notification_text = f'{notification_attrs}, {notification_data}'
+    notification_attr = ''
+    if 'attributes' in pubsub_message:
+        notification_attr = json.dumps(pubsub_message['attributes'])
 
-    slack_client.api_call(
-      'chat.postMessage',
-      channel=CHANNEL_ID,
-      text=budget_notification_text)
+    notification_data = ''
+    if 'data' in pubsub_message:
+        notification_data = base64.b64decode(data['data']).decode('utf-8')
+
+    budget_notification_text = f'{notification_attr}, {notification_data}'
+
+    try:
+        slack_client.api_call(
+            'chat.postMessage',
+            json={
+                'channel': CHANNEL,
+                'text'   : budget_notification_text
+            }
+        )
+    except Exception:
```

Comment:
Can you specify an exception to catch? I worry that using the general `Exception` is too broad and may lead to poor DevX if things go wrong

Reply:
I think technically it would be SlackApiError, but I don't know the full details of the scope of Slack's API and the types of errors that may be thrown.

Reply:
I'll specify SlackAPIError